rma/models/rma.py

```diff
@@ -0,0 +1,954 @@
+                raise ValidationError(
+                    _("This RMA cannot be returned to the customer."))
+        else:
+            if not self.filtered("can_be_delivered"):
+                raise ValidationError(
+                    _("None of the selected RMAs can be returned "
+                      "to the customer."))
+
+    def _ensure_can_be_split(self):
+        """intended to be called before launch and after save the split wizard.
+        invoked by:
+        rma.action_split
+        rma.extract_quantity
+        """
+        self.ensure_one()
+        if not self.can_be_split:
+            raise ValidationError(_("This RMA cannot be split"))
+
+    def _ensure_required_fields(self):
+        """ This method is used to ensure the following fields are not empty:
+        ['partner_id', 'partner_invoice_id', 'product_id']
+
+        This method is intended to be called on confirm RMA action and is
+        invoked by:
+        rma._check_required_after_draft
+        rma.action_confirm
+        """
+        ir_translation = self.env['ir.translation']
+        required = ['partner_id', 'partner_invoice_id', 'product_id']
+        for record in self:
+            desc = ""
+            for field in filter(lambda item: not record[item], required):
+                desc += "\n%s" % ir_translation.get_field_string("rma")[field]
+            if desc:
+                raise ValidationError(_('Required field(s):%s') % desc)
+
+    def _ensure_valid_qty_to_extract(self, qty, uom):
+        to_split_uom_qty = qty
+        if uom != self.product_uom:
+            to_split_uom_qty = uom._compute_quantity(qty, self.product_uom)
+        if qty > 0.0 and to_split_uom_qty > self.remaining_qty:
+            raise ValueError(
+                _("Quantity to extract cannot be greater than remaining "
+                  "delivery quantity (%s %s)")
+                % (self.remaining_qty, self.product_uom.name)
+            )
+
+    # Reception business methods
+    def _create_receptions_from_picking(self):
+        self.ensure_one()
+        return_wizard = self.env['stock.return.picking'].with_context(
+            active_id=self.picking_id.id,
+            active_ids=self.picking_id.ids,
+        ).create({})
+        return_wizard.product_return_moves.filtered(
+            lambda r: r.move_id != self.move_id
+        ).unlink()
+        return_line = return_wizard.product_return_moves
+        return_line.quantity = self.product_uom_qty
+        picking_action = return_wizard.create_returns()
+        picking_id = picking_action['res_id']
+        picking = self.env['stock.picking'].browse(picking_id)
+        picking.origin = "%s (%s)" % (self.name, picking.origin)
+        move = picking.move_lines
+        move.priority = self.priority
+        return move
+
+    def _create_receptions_from_product(self):
+        self.ensure_one()
+        incoming_picking_type = self.env['stock.picking.type'].search(
```

Comment:
For this case it seems quite trivial to have an operation_type to choose in the rma and default to these values if it isn't set